ext/dom/document.c

```diff
@@ -1484,42 +1485,57 @@ PHP_METHOD(DOMDocument, saveXML)
 			php_dom_throw_error(WRONG_DOCUMENT_ERR, dom_get_strict_error(intern->document));
 			RETURN_FALSE;
 		}
+
 		buf = xmlBufferCreate();
 		if (!buf) {
 			php_error_docref(NULL, E_WARNING, "Could not fetch buffer");
 			RETURN_FALSE;
 		}
-		if (options & LIBXML_SAVE_NOEMPTYTAG) {
-			saveempty = xmlSaveNoEmptyTags;
-			xmlSaveNoEmptyTags = 1;
-		}
+		saveempty = xmlSaveNoEmptyTags;
+		xmlSaveNoEmptyTags = !!(options & LIBXML_SAVE_NOEMPTYTAG);
 		xmlNodeDump(buf, docp, node, 0, format);
-		if (options & LIBXML_SAVE_NOEMPTYTAG) {
-			xmlSaveNoEmptyTags = saveempty;
-		}
-		mem = (xmlChar*) xmlBufferContent(buf);
-		if (!mem) {
-			xmlBufferFree(buf);
+		xmlSaveNoEmptyTags = saveempty;
```

Comment:
Is ``xmlSaveNoEmptyTags`` a global libxml2 variable? Because that code makes no sense otherwise

So maybe add some comment about this?

Reply:
> Is xmlSaveNoEmptyTags a global libxml2 variable?

Yes

> So maybe add some comment about this?

Will do